src/coord/axisCommonTypes.ts

```diff
@@ -45,8 +45,9 @@ export interface AxisBaseOptionCommon extends ComponentOption,
         placeholder?: string;
     };
     nameTextStyle?: AxisNameTextStyleOption;
-    // The gap between axisName and axisLine.
+    // The gap between axisName and axisLine/labels
     nameGap?: number;
+    nameLayout?: 'auto';
```

Comment:
An alternative was adding `'auto'` as one of the `nameGap`'s allowed values, but that would take away the ability to control the automatically managed distance between the name and the labels.